### PR TITLE
Updated to Ruby 2.2.3. Swapped out Unicorn for Puma.

### PR DESCRIPTION
**Pull request by jmccartie**
After seeing timeouts on this app on Heroku, decided to swap out for Puma to allow for more concurrency. It's the [recommend web server on Heroku](https://devcenter.heroku.com/articles/ruby-default-web-server#production-web-server) and works very well for my other apps.

Decided to upgrade to Ruby 2.2 as well since 2.1 has some default memory behavior that needs tuning on Heroku. 

In order to get the tests to pass, I had to make some DBCleaner changes, but even then, the test suite runs very slow -- primarily because of the capybara-webkit usage. This might be an area to improve on in the future.

### CODE REVIEW

**Comment by jmccartie** [comment]
`database.yml` was put back in since there is new default behavior here on Heroku with Rails 4.1. I think having it committed -- but with ENV vars -- is a better practice than an example file.

File: Gemfile [hunk 1]
```diff
@@ -1,7 +1,9 @@
 source 'https://rubygems.org'
-ruby '2.1.2'
+ruby '2.2.3'
 
 gem 'rails', '4.1.2'
+gem 'puma', '~> 2.13.4'
```
Comment (evanphx) [inline Gemfile]: This should be '~> 2.13' so that it picks up '2.14' and so on.

Comment (jmccartie) [inline Gemfile]: Agreed. I'll make another PR.

Comment (mghaught) [inline Gemfile]: I can fix in my testing

Comment (jmccartie) [inline Gemfile]: Done.  #52

**Pull request merged**